lib/Util.ts

```diff
@@ -4,9 +4,10 @@ import * as url from 'url';
  * creates an url from a request url and optional base url (http://server:8080)
  * @param {string} requestUrl - a fully qualified url or relative url
  * @param {string} baseUrl - an optional baseUrl (http://server:8080)
+ * @param {boolean} preservePath - an optional baseUrl (http://server:8080)
  * @return {string} - resultant url 
  */
-export function getUrl(requestUrl: string, baseUrl?: string): string  {
+export function getUrl(requestUrl: string, baseUrl?: string, preservePath?: boolean): string  {
```

Comment:
I'm struggling with whether this should be the default.  In the normal case of just providing the hostname portion (no path) it does the right thing.  In the case where the base url has a relative path, it does the right thing.  I need to think about it.  

This is pre-release about to go to 1.0 and this change should bump the minor version (so ^ with patch level won't drift into this behavior) and we can doc.

The other thought was flip to opt in to chop the relative path but if that's your intent, just provide the hostname portion.  No need for another optional arg ...hmmmm

Thoughts?

Reply:
Since this is still prerelease, my preference would be to never strip the relative path. Reason being similar to your thought about flipping to opt in. If the use case comes up that the base URL also has a path, then it would make sense to explicitly state it in the baseUrl parameter, and always expect it to be there. At that point the optional parameter can be completely removed. That would make this a breaking change though...

Reply:
@kaeedo Could you add some examples of usage for this feature? Then we can speak to the explicit use cases. It often helps think things through and make the "right" choice more apparent. (based on the perspective of a consumer of the library). As well as what the consumer would have to do if we didn't have this feature.

It's tricky to balance where we draw the line of what's included and what isn't.

Reply:
@kaeedo - I agree on your points.  Bump the minor version with this change.  That mitigates the pre-release breaking change since folks default to sliding on patch versions.  And it's only a change when you're doing something non mainstream (passing base url with path but then expecting it not to be honored).  Let's move forward without the arg.  Add the test cases on trailing and leading slashes.

Reply:
@stephenmichaelf 
For the use case in my office, we have a reverse proxy configured to point to different docker containers:

    https://www.contoso.com/firstcontainer
    https://www.contoso.com/secondcontainer
    https://www.contoso.com/thirdcontainer

I would like to create an instance of a `RestClient` such that `https://www.contoso.com/firstcontainer` is the `baseUrl` instead of just `https://www.contoso.com` and then having to prepend `firstcontainer` to every rest request